### PR TITLE
Update our disabilities question page in our hesa questionnaire

### PR DESCRIPTION
**Pull request by tomas-stefano**
## Context

This PR adds the page that candidates can enter disabilities or choose between the options.

## What is changed?

Screenshot 2022-10-07 at 09 29 36

### CODE REVIEW

**Comment by tomas-stefano** [comment]
[[Needs to be done by 11 Oct] Update our disabilities question page in our HESA questionnaire](https://trello.com/c/jez3Cw5A/732-needs-to-be-done-by-11-oct-update-our-disabilities-question-page-in-our-hesa-questionnaire)

**Comment by tomas-stefano** [comment]
For those helping to test, we need to test in the review app:

* A new application
* Carry over application with the disabilities "Blind" or "Deaf" etc
* Carry over application with Other disability.

And go and submit an application until it reaches the disabilities or check your answers page.

File: app/views/candidate_interface/equality_and_diversity/edit_disabilities.html.erb [hunk 1]
```diff
@@ -1,19 +1,31 @@
...
       <%= f.govuk_error_summary %>
 
       <%= f.govuk_check_boxes_fieldset :disabilities, multiple: false, caption: { text: t('equality_and_diversity.title'), size: 'xl' }, legend: { text: t('equality_and_diversity.disabilities.title'), size: 'xl', tag: 'h1' } do %>
+        <p class="govuk-body">Training providers will not see your answer to this question when deciding whether to offer you a place.</p>
+
+        <p class="govuk-body">They will only see your answer if you accept an offer from them.</p>
+
+        <p class="govuk-body">
+          You can <%= govuk_link_to 'ask for support as part of your application', candidate_interface_new_training_with_a_disability_path %> if you want to tell training providers about your needs straight away.</p>
+
```
Comment (frankieroberto) [inline app/views/candidate_interface/equality_and_diversity/edit_disabilities.html.erb:16]: The paragraphs should come before the fieldset, and there should be a separate h1 ("Disabilities and health conditions") from the legend ("Do you have any of these disabilities or health conditions?").

File: app/views/candidate_interface/equality_and_diversity/edit_disabilities.html.erb [hunk 2]
```diff
@@ -1,19 +1,31 @@
...
     <%= form_with model: @disabilities, url: candidate_interface_edit_equality_and_diversity_disabilities_path, method: :patch do |f| %>
       <%= f.govuk_error_summary %>
 
       <%= f.govuk_check_boxes_fieldset :disabilities, multiple: false, caption: { text: t('equality_and_diversity.title'), size: 'xl' }, legend: { text: t('equality_and_diversity.disabilities.title'), size: 'xl', tag: 'h1' } do %>
+        <p class="govuk-body">Training providers will not see your answer to this question when deciding whether to offer you a place.</p>
+
+        <p class="govuk-body">They will only see your answer if you accept an offer from them.</p>
+
+        <p class="govuk-body">
+          You can <%= govuk_link_to 'ask for support as part of your application', candidate_interface_new_training_with_a_disability_path %> if you want to tell training providers about your needs straight away.</p>
```
Comment (frankieroberto) [inline app/views/candidate_interface/equality_and_diversity/edit_disabilities.html.erb:15]: This link seems to take you to a page where your previous answer isn’t pre-populated, for some reason? I think maybe it needs to take you to the edit view?

**Review by frankieroberto — CHANGES_REQUESTED** [review]
Tested in review app. A couple of changes needed - mainly adding a separate h1 from the legend.

There are some changes to the Check your answers page needed, but that can be a separate PR.

File: app/forms/candidate_interface/equality_and_diversity/disabilities_form.rb [hunk 3]
```diff
@@ -9,11 +12,11 @@ class EqualityAndDiversity::DisabilitiesForm
     def self.build_from_application(application_form)
       return new(disabilities: nil) if application_form.equality_and_diversity.nil?
 
-      list_of_disabilities = DisabilityHelper::STANDARD_DISABILITIES.map { |_, disability| disability } << 'Other'
-      listed, other = application_form.equality_and_diversity['disabilities'].partition { |d| list_of_disabilities.include?(d) }
+      list_of_disabilities = DisabilityHelper::STANDARD_DISABILITIES.map { |_, disability| disability } << OTHER << OPT_OUT << NONE
```
Comment (carlosmartinez) [inline app/forms/candidate_interface/equality_and_diversity/disabilities_form.rb:15]: i think more intuitive:
```
      list_of_disabilities = DisabilityHelper::STANDARD_DISABILITIES.map { |_, disability| disability } + [OTHER, OPT_OUT,  NONE]
```

File: spec/factories/application_form.rb [hunk 4]
```diff
@@ -61,6 +61,10 @@
         ethnicity = Class.new.extend(EthnicBackgroundHelper).all_combinations.sample
         other_disability = 'Acquired brain injury'
         all_disabilities = DisabilityHelper::STANDARD_DISABILITIES.map(&:second) << other_disability
+        if RecruitmentCycle.current_year < 2023
```
Comment (carlosmartinez) [inline spec/factories/application_form.rb:64]: can we replace `2023` with some meaningfully-named constant?

**Comment by frankieroberto** [comment]
Can the error message when not checking any checkbox be updated to this please:

> Select any disabilities or health conditions you have, or select ‘I do not have any’ or ‘Prefer not to say’

(Sorry for the last-minute request!)

File: app/views/candidate_interface/equality_and_diversity/edit_disabilities.html.erb (hunk 1 above)
Comment (tomas-stefano) [reply app/views/candidate_interface/equality_and_diversity/edit_disabilities.html.erb]: Done!

**Review by frankieroberto — APPROVED** [review]

**Pull request merged**